src/main/java/ProjectOcean/Controller/CourseController.java

```diff
@@ -115,27 +122,42 @@ public void relocateToPoint(Point2D p) {
 
     @FXML
     private void dragDetected(MouseEvent event) {
-        //Put a copy of the object that was dragged in the Clipboard to enable drag and drop.
-        CourseListIconController icon = (CourseListIconController) event.getSource();
-        ClipboardContent content = new ClipboardContent();
-        content.putString(icon.toString());
 
 
-        //Check from which parent the object started in.
-        switch (icon.getParent().getId()){
+      //  CourseController draggedObject = getICourse();
+        owner = this.getParent();
+        copyDraggedObjectToClipBoard(this);
+
+
+
+        //Check from which parent the object started in and delete from the model.
+        switch (owner.getId()){
             case "workspaceContainer":
-                model.removeCourseFromWorkspace(icon.getICourse());
+                model.removeCourseFromWorkspace(course);
                 break;
+            case "yearGrid":
+                //TODO Come back and fix a better solution.
+                YearController yearController = (YearController) owner.getParent().getParent().getParent();
+                yearController.removeCourse(course);
+                //  model.removeCourse();
```

Comment:
Some commented code. Would it be possible to put "coursesInYear" in the model instead to avoid the dependency on controller? removeCourse in YearController needs this variable so removeCourse would then be put in model too.

Reply:
Would be kind of strange to put the map in the model in the class year. Since we would be storing the same data in two different ways. I certainly see the problem with double dependency though. We should probably use an interface. If you let it through now we can tackle it when we try tackle the "too many interface" problem.